lib/modules/product.rb

```diff
@@ -0,0 +1,30 @@
+module Store
+  class Product
+    attr_reader :id, :name, :price, :vat
+
+    @@last_id = 0
+    
+    def initialize(name, price, vat = 23)
+      @id = next_id
+      @name = validate(name, String)
+      @price = validate(price, Numeric)
+      @vat = validate(vat, Numeric)
+    end
+
+    def next_id
+      @@last_id += 1
+    end
+
+    def validate(input, type)
+      unless input.nil?
+        if input.is_a? type then return input else raise TypeError end
```

Comment:
This should also work: `input.is_a?(type) ? input : raise TypeError`. Do not use `if - then` version.